es6-lib/decoders/kmz.js

```diff
@@ -78,31 +75,42 @@ class KMZ extends Duplex {
             kmlStream.pause();
           }
         }
-      })
-      .on('end', () => {
-        logger.info('Done reading KML stream');
-        if (!this._readableState.ended) this.push(null);
       });
   }
 
+  _onError(err) {
+    this.emit('error', err);
+  }
+
   _startPushing() {
     this._isPushing = true;
     var hasOpened = false;
```

Comment:
no it's used, i added it to remove the duplicated `(err) => this.emit('error', err)` stuff littering the streams